doc/SONiC_VRRP_HLD.md

```diff
@@ -0,0 +1,582 @@
+| ARP           | Address Resolution Protocol                                  |
+| CLI           | Command Line Interface                                       |
+| IP            | Internet Protocol                                            |
+| LAG           | Link Aggregation Group                                       |
+| LAN           | Local Area Network                                           |
+| MAC           | Media Access Control addresses                               |
+| VIP           | Virtual IP address                                           |
+| VLAN          | Virtual Local Area Network                                   |
+| VMAC          | Virtual MAC address                                          |
+| VRF           | Virtual Routing and Forwarding                               |
+| VRID          | Virtual Router Identifier                                    |
+| VRRP          | Virtual Router Redundency Protocol                           |
+| VRRP Instance | An instance of VRRP state machine on an interface. Multiple VRRP state machines can be configured on an interface. |
+| VRRP Owner    | VRRP owner (of a VRRP instance) is the router whose virtual IP address is the same as the real interface IP address |
+
+# About this Manual
+
+This document provides general overview of VRRP feature implementation based on RFC 3768 in SONiC. 
+
+
+
+# 1 Introduction and Scope
+
+This document describes the Functionality and High level design of the VRRP feature.
+
+The Virtual Router Redundancy Protocol (VRRP) functionality is designed to eliminate the single point of
+failure inherent in the static default routed environment. VRRP specifies an election protocol that
+dynamically assigns responsibility of gateway router to a VRRP instance on one of the routers on a LAN. The VRRP instance controlling the IP address(es) associated with a virtual router is called the Master, and routes the traffic. The election process provides dynamic fail-over in the forwarding responsibility should the Master become unavailable. Any of the virtual router's IP addresses on a LAN can then be used as the default first hop router by end-hosts. The advantage gained from using VRRP is a higher availability default path without requiring configuration of dynamic routing or router discovery protocols on every end-host.
+
+
+
+
+# 2 Feature Requirements
+
+## 2.1 Functional Requirements
+
+Following requirements are addressed by the design presented in this document:
+
+1. Support VRRP v2 (IPv4) 
+
+2. Support multiple VRRP instances (groups) per interface
+
+3. Support VRRP on VLAN, PortChannel and Ethernet interfaces
+
+4. Support Uplink interface tracking
+
+5. Support pre-emption of a Master when a high priority VRRP node comes up
+
+6. Support configurable priority for VRRP instance
+
+7. Support configuration and management of various VRRP parameters
+
+    
+
+Following requirements are beyond scope of this release. 
+
+1. VRRPv3 (IPv6) support
```

Comment:
what is preventing not to support vrrp3? 
How is it different from FRR VRRP? Since FRR is part of SONiC stack, can't we stick to FRR VRRP (one stack)?

Reply:
Currently VRRPv3 (IPv6) support is added.

Since keepalived VRRP is robust, feature rich and well maintained in linux community, keepalived VRRP was chosen over FRR VRRP.